machine/v1/types_powervsprovider.go

```diff
@@ -190,3 +204,18 @@ type PowerVSSecretReference struct {
 	// +optional
 	Name string `json:"name,omitempty"`
 }
+
+// LoadBalancerReference is a reference to a load balancer on IBM Cloud virtual private cloud(VPC).
+type LoadBalancerReference struct {
+	// Name of the LoadBalancer in IBM Cloud VPC
+	// It is a reference to existing LoadBalancer created by openshift installer component.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
```

Comment:
Should start with the json tag name not the Go field name, is there any way to restrict this? For example can we explain what character set, minimum/maximum lengths are required for this field?
```suggestion
	// name of the LoadBalancer in IBM Cloud VPC
	// It is a reference to existing LoadBalancer created by openshift installer component.
	// +kubebuilder:validation:Required
	Name string `json:"name"`
```

Reply:
sure , not sure how can we restrict it, Currently the loadbalancer name will be of format "{cluster-id}-loadbalancer" for public and "{cluster-id}-loadbalancer-int" for internal loadbalancer

Reply:
Is cluster-id restricted at all by the installer? We can probably restrict this to alphanumeric characters, hyphens and underscores, but we should check what restrictions are imposed by the installer

Reply:
I could see these restrictions: https://github.com/Karthik-K-N/installer/blob/e91df626c10e569e7613249053b7b9b264db42df/pkg/asset/installconfig/clusterid.go#L37-L50

Reply:
Apologies we are using infraID not clusterid, Infra ID is generated like this: https://github.com/Karthik-K-N/installer/blob/e91df626c10e569e7613249053b7b9b264db42df/pkg/asset/installconfig/clusterid.go#L60-L79

I think we can add validation for alphanumeric with hyphen,

Reply:
Yeah looks like it can be limited to alphanum and hyphen, great!